template/content/clients.js

```diff
@@ -0,0 +1,141 @@
+'use strict';
+
+var TChannel = require('tchannel');
+var HyperbahnClient = require('tchannel/hyperbahn');
+var fs = require('fs');
+var path = require('path');
+var Ringpop = require('ringpop');
+var Logtron = require('logtron');
+var Statsd = require('uber-statsd-client');
+var os = require('os');
+var process = require('process');
+var myLocalIp = require('my-local-ip');
+var Uncaught = require('uncaught-exception');
+
+var thriftFile = fs.readFileSync(
+    path.join(__dirname, 'thrift', 'service.thrift'), 'utf8'
+);
+
+var SERVICE_NAME = 'my-service';
+
+/* TODO:
+    - process reporter
+    - heap dump
+    - repl
+*/
+module.exports = ApplicationClients;
+
+function ApplicationClients(config, options) {
```

Comment:
This is a pretty busy thing; totally agree;

However I tried separating it out and its just spreads the mess around. I think keeping it all in one monolithic file is a good idea as this is really the entire glue layer.

Spreading the complexity out just hides it.